department-of-veterans-affairs/vets-api
# [64191] Add Betamock Configuration for MAP STS

## Summary

- This PR adds the MAP STS endpoint details to the betamocks configuration.
- This facilitates for rapid development for ToU and other features interacting with MAP STS API; otherwise, development requires use of SOCKS proxy.
- This work is being completed by @vsp-identity

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#64191

## Testing done

Prior to this change, any call to the STS server would fail. This PR enables betamocks for these calls, so they should succeed and return a mocked response.

### Test Steps

- Pull the latest changes to the `vets-api-mockdata` repo.
  - Changes should include additions from PR https://github.com/department-of-veterans-affairs/vets-api-mockdata/pull/417
  - If this PR isn't merged yet, check out the branch locally.
- Open a rails console.
- Require the STS service.
```ruby
require 'mobile_application_platform/security_token/service'
```
- Call the service with a valid application parameter and any value for ICN:
```ruby
MobileApplicationPlatform::SecurityToken::Service.new.token(application: :sign_up_service, icn: 'bar')
```
- Verify the response is mocked.

## Screenshots

Not relevant.

## What areas of the site does it impact?

This does not have an impact to production facilities of the site; it only affects local development.

## Acceptance criteria


## Requested Feedback

Please advise on best practices using the betamocks library and whether additional configuration is necessary.
